Flow.Launcher.Infrastructure/Storage/JsonStorage.cs

```diff
@@ -11,62 +12,73 @@ namespace Flow.Launcher.Infrastructure.Storage
     /// </summary>
     public class JsonStorage<T> where T : new()
     {
-        protected T _data;
+        protected T? Data;
         // need a new directory name
         public const string DirectoryName = "Settings";
         public const string FileSuffix = ".json";
-        public string FilePath { get; set; }
-        public string DirectoryPath { get; set; }
+        protected string FilePath { get; init; } = null!;
+        private string TempFilePath => $"{FilePath}.tmp";
+        private string BackupFilePath => $"{FilePath}.bak";
+        protected string DirectoryPath { get; init; } = null!;
 
 
         public T Load()
         {
+            string? serialized = null;
+
             if (File.Exists(FilePath))
             {
-                var serialized = File.ReadAllText(FilePath);
-                if (!string.IsNullOrWhiteSpace(serialized))
+                serialized = File.ReadAllText(FilePath);
+            }
+
+            if (!string.IsNullOrEmpty(serialized))
+            {
+                try
                 {
-                    Deserialize(serialized);
+                    Data = JsonSerializer.Deserialize<T>(serialized)?? TryLoadBackup() ?? LoadDefault();
                 }
-                else
+                catch (JsonException)
                 {
-                    LoadDefault();
+                    Data = TryLoadBackup() ?? LoadDefault();
                 }
             }
             else
             {
-                LoadDefault();
+                Data = TryLoadBackup() ?? LoadDefault();
             }
-            return _data.NonNull();
+            return Data.NonNull();
         }
 
-        private void Deserialize(string serialized)
+        private T LoadDefault()
         {
-            try
-            {
-                _data = JsonSerializer.Deserialize<T>(serialized);
-            }
-            catch (JsonException e)
+            if (File.Exists(FilePath))
             {
-                LoadDefault();
-                Log.Exception($"|JsonStorage.Deserialize|Deserialize error for json <{FilePath}>", e);
+                BackupOriginFile();
             }
 
-            if (_data == null)
-            {
-                LoadDefault();
-            }
+            return new T();
         }
 
-        private void LoadDefault()
+        private T? TryLoadBackup()
         {
-            if (File.Exists(FilePath))
+            if (!File.Exists(BackupFilePath))
+                return default;
+
+            try
             {
-                BackupOriginFile();
+                var data = JsonSerializer.Deserialize<T>(File.ReadAllText(BackupFilePath));
+                if (data != null)
+                {
+                    Log.Info($"|JsonStorage.Load|Load backup file {BackupFilePath} successfully");
```

Comment:
Should this message be failed to load settings.json, back up file loaded successfully?

Reply:
```suggestion
                    Log.Info($"|JsonStorage.Load|Failed to load settings.json, {BackupFilePath} restored successfully");
```